addons/point_of_sale/models/res_partner.py

```diff
@@ -7,6 +7,11 @@ class ResPartner(models.Model):
     _inherit = 'res.partner'
 
     barcode = fields.Char(string='Barcode', help="BarCode", oldname='ean13')
+    pos_order_count = fields.Integer(compute='_compute_pos_order', help="The number of point of sale orders related to this customer")
+
+    def _compute_pos_order(self):
+        for partner in self:
+            partner.pos_order_count = self.env['pos.order'].search_count([('partner_id', '=', partner.id)])
```

Comment:
Instead of calling search_count on each partner on self, use read_group instead ==> 1 call, same result